### PR TITLE
ATLAS-168: Notify users when their marker is fading

### PR DESCRIPTION
**Pull request by HelioStrike**
JIRA issue: https://issues.openmrs.org/projects/ATLAS/issues/ATLAS-168?filter=allissues

--- WORK IN PROGRESS ---

See [this](https://talk.openmrs.org/t/gsoc-2019-openmrs-atlas-3-1/22226/175).

### CODE REVIEW

**Comment by HelioStrike** [comment]
@bmamlin I updated the PR. What do you think of the new tables? I haven't written code to handle the email links, but I'll try to finish it tomorrow.

File: db/01_database.sql [hunk 1]
```diff
@@ -184,6 +184,40 @@ CREATE TABLE `rss` (
...
+-- Table structure for table `notifications`
+--
+
+DROP TABLE IF EXISTS `notifications`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `notifications` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `marker_id` varchar(38) COLLATE utf8_unicode_ci NOT NULL,
+  `principal` varchar(255) COLLATE utf8_unicode_ci NOT NULL,
```
Comment (bmamlin) [inline db/01_database.sql]: `principal` is a security term used in the `auth` table to cover authentication of users & non-users. For notifications & unsubscribed, I'd suggest using `username`.

File: db/01_database.sql [hunk 2]
```diff
@@ -184,6 +184,40 @@ CREATE TABLE `rss` (
...
+-- Table structure for table `unsubscribed`
+--
+
+DROP TABLE IF EXISTS `unsubscribed`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `unsubscribed` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `marker_id` varchar(38) COLLATE utf8_unicode_ci NOT NULL,
+  `principal` varchar(255) COLLATE utf8_unicode_ci NOT NULL,
```
Comment (bmamlin) [inline db/01_database.sql]: Use `username` instead of `principal` per comment above.

**Comment by HelioStrike** [comment]
@bmamlin Updated the PR with all the changes suggested on Talk. I tested it on stg (
with a much lower mail timeout), and it seems to be working fine.

Screenshot from 2019-07-22 21-26-30

**Pull request merged**